tests/components/monzo/test_config_flow.py

```diff
@@ -136,3 +141,155 @@ async def test_config_non_unique_profile(
     result = await hass.config_entries.flow.async_configure(result["flow_id"])
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+async def test_config_reauth_profile(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    polling_config_entry: MockConfigEntry,
+    monzo: AsyncMock,
+    current_request_with_host: None,
+) -> None:
+    """Test reauth an existing profile reauthenticates the config entry."""
+    await setup_integration(hass, polling_config_entry)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": SOURCE_REAUTH,
+            "entry_id": polling_config_entry.entry_id,
+        },
+        data=polling_config_entry.data,
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+    assert result["url"] == (
+        f"{OAUTH2_AUTHORIZE}/?"
+        f"response_type=code&client_id={CLIENT_ID}&"
+        "redirect_uri=https://example.com/auth/external/callback&"
+        f"state={state}"
+    )
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.clear_requests()
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "refresh_token": "mock-refresh-token",
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+            "user_id": str(USER_ID),
+        },
+    )
+    with patch(
+        "homeassistant.config_entries.ConfigEntries.async_update_entry",
+        return_value=True,
+    ) as mock_setup:
```

Comment:
I _think_ I made this a fixture in conftest.py, so you can just add that to the parameters of the test

Reply:
I can't see an async_update_entry mock in tests.conftest or tests.components.conftest - it might've been something else?

Reply:
Maybe that's a nice one to add as well, makes it cleaner :)

Reply:
oh wait, I looked wrong

Reply:
We shouldn't patch this at all, I thought this was async_setup_entry